timelock-server/src/test/java/com/palantir/atlasdb/timelock/clock/ClockSkewMonitorIntegrationTest.java

```diff
@@ -0,0 +1,122 @@
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.timelock.clock;
+
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import static com.palantir.atlasdb.timelock.clock.ClockSkewMonitor.PAUSE_BETWEEN_REQUESTS;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.jmock.lib.concurrent.DeterministicScheduler;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.internal.util.collections.Sets;
+
+import com.google.common.collect.Maps;
+
+public class ClockSkewMonitorIntegrationTest {
+    private final String server = "test";
+    private final ClockService mockedLocalTimeSupplier = mock(ClockService.class);
+    private final ClockService mockedRemoteClockService = mock(ClockService.class);
+
+    private ClockSkewEvents mockedEvents;
+    private final Map<String, ClockService> monitorByServer = new HashMap<>();
+    private final Map<String, RequestTime> previousRequestsByServer = new HashMap<>();
+
+    @SuppressWarnings("unchecked")
+    private DeterministicScheduler executorService = new DeterministicScheduler();
+    private ClockSkewMonitor monitor;
+
+    @Before
+    public void setUp() {
+        mockedEvents = mock(ClockSkewEvents.class);
+        monitorByServer.put(server, mockedRemoteClockService);
+        previousRequestsByServer.put(server, RequestTime.EMPTY);
+        monitor = new ClockSkewMonitor(monitorByServer, previousRequestsByServer,
+                mockedEvents, executorService, mockedLocalTimeSupplier);
+    }
+
+    @Test
+    public void logsRequestsWithoutSkew() {
+        monitor.runInBackground();
+        RequestTime requestTime = new RequestTime(1L, 1L, 1L);
+        mockLocalAndRemoteClockSuppliers(requestTime);
+        executorService.tick(1, TimeUnit.NANOSECONDS);
+
+        RequestTime remoteTime = new RequestTimeBuilder(requestTime)
+                .progressLocalClock(100L)
+                .progressRemoteClock(100L)
+                .build();
+        mockLocalAndRemoteClockSuppliers(remoteTime);
+        executorService.tick(PAUSE_BETWEEN_REQUESTS.toNanos(), TimeUnit.NANOSECONDS);
+
+        verify(mockedEvents, times(1))
+                .requestPace(anyString(), anyLong(), anyLong(), anyLong());
```

Comment:
might be better to have this test if the thing actually logs skew, since that's the critical feature